src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java

```diff
@@ -259,6 +270,137 @@ public static void revokeIdTokens(
+      NewCookie cookie = new NewCookie("session", sessionCookie /* ... other parameters */);
+      return Response.ok().cookie(cookie).build();
+    } catch (Exception e) {
+      return Response.status(Status.UNAUTHORIZED).entity("Failed to create a session cookie")
+          .build();
+    }
+  }
+  // [END session_login]
+
+  public Response checkAuthTime(String idToken) throws Exception {
+    // [START check_auth_time]
+    // To ensure that cookies are set only on recently signed in users, check auth_time in
+    // ID token before creating a cookie.
+    FirebaseToken decodedToken = FirebaseAuth.getInstance().verifyIdTokenAsync(idToken).get();
+    long authTimeMillis = (long) decodedToken.getClaims().get("auth_time") * 1000L;
+
+    // Only process if the user signed in within the last 5 minutes.
+    if (System.currentTimeMillis() - authTimeMillis < TimeUnit.MINUTES.toMillis(5)) {
+      long expiresIn = TimeUnit.DAYS.toMillis(5);
+      SessionCookieOptions options = SessionCookieOptions.builder()
+          .setExpiresIn(expiresIn)
+          .build();
+      String sessionCookie = FirebaseAuth.getInstance().createSessionCookieAsync(
+          idToken, options).get();
+      // Set cookie policy parameters as required.
+      NewCookie cookie = new NewCookie("session", sessionCookie);
+      return Response.ok().cookie(cookie).build();
+    }
+    // User did not sign in recently. To guard against ID token theft, require
+    // re-authentication.
+    return Response.status(Status.UNAUTHORIZED).entity("Recent sign in required").build();
+    // [END check_auth_time]
+  }
+
+  private Response serveContentForUser(FirebaseToken decodedToken) {
+    return null;
+  }
 
+  private Response serveContentForAdmin(FirebaseToken decodedToken) {
+    return null;
+  }
+
+  // [START session_verify]
+  @POST
+  @Path("/profile")
+  public Response verifySessionCookie(@CookieParam("session") Cookie cookie) {
+    String sessionCookie = cookie.getValue();
+    try {
+      // Verify the session cookie. In this case an additional check is added to detect
+      // if the user's Firebase session was revoked, user deleted/disabled, etc.
+      final boolean checkRevoked = true;
+      FirebaseToken decodedToken = FirebaseAuth.getInstance().verifySessionCookieAsync(
+          sessionCookie, checkRevoked).get();
+      return serveContentForUser(decodedToken);
+    } catch (Exception e) {
+      // Session cookie is unavailable, invalid or revoked. Force user to login.
+      return Response.temporaryRedirect(URI.create("/login")).build();
+    }
+  }
+  // [END session_verify]
+
+  public Response checkPermissions(String sessionCookie) {
+    // [START session_verify_with_permission_check]
+    try {
+      final boolean checkRevoked = true;
+      FirebaseToken decodedToken = FirebaseAuth.getInstance().verifySessionCookieAsync(
+          sessionCookie, checkRevoked).get();
+      if (Boolean.TRUE.equals(decodedToken.getClaims().get("admin"))) {
```

Comment:
Does Java unboxing allow to make this just `if(decodedToken.getClaims().get("admin"))`?